specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json

```diff
@@ -4273,6 +4274,14 @@
             "modelAsString": true
           },
           "description": "Test coverage for the endpoint."
+        },
+        "locationDetails": {
+          "$ref": "#/definitions/ConnectionMonitorEndpointLocationDetails",
+          "description": "Location details for connection monitor endpoint."
+        },
+        "subscriptionId": {
+          "type": "string",
+          "description": "Subscription ID for connection monitor endpoint."
         }
```

Comment:
its not the subscription ID of the connection monitor but the monitored endpoint? Which may be in some other subscription? 

Why is this needed can it be the same as resourceID?

Reply:
Is this only for the ARC scenario? A: YES

Reply:
You mentioned public docs already doc this, I don't see how that is possible using proper process, please share a link.

Reply:
> its not the subscription ID of the connection monitor but the monitored endpoint? Which may be in some other subscription?
> 
> Why is this needed can it be the same as resourceID?

As you can see, its related to specific endpoint type of connection monitor ,
for "AzureArcNetwork" like other endpoints (-AzureVM/-ExternalAddress/-AzureArcVM and others).
for info [link](https://learn.microsoft.com/en-us/powershell/module/az.network/new-aznetworkwatcherconnectionmonitorendpointobject?view=azps-11.1.0)
For this endpoint type "AzureArcNetwork", we need some validation that requires this subscriptionId.

Reply:
> Is this only for the ARC scenario? A: YES

Yes, for "AzureArcNetwork" type.

Reply:
> You mentioned public docs already doc this, I don't see how that is possible using proper process, please share a link.

Not able to understand this query, I am not sure if you are looking for [this](https://learn.microsoft.com/en-us/powershell/module/az.network/new-aznetworkwatcherconnectionmonitorendpointobject?view=azps-11.1.0).

Reply:
subscriptionId is not of the connection monitor's subscriptionId.
 
To monitors onpremise machines, customer has to setup arc machines. In order to monitor arc machines from different subscription, this field is being used.
 
Changes are already made in network resource provider using which connection-monitors can be created with ArcNetwork as one endpoint,  but that is not publicly documented currently. It's in progress right now.
https://learn.microsoft.com/en-us/rest/api/network-watcher/connection-monitors/create-or-update?view=rest-network-watcher-2023-06-01&tabs=HTTP#endpointtype